lib/VM/JSLib/DateUtil.cpp

```diff
@@ -972,15 +1045,42 @@ static double parseESDate(StringView str) {
   else
     return nan;
 
-  // Hour and minute of timezone adjustment.
-  if (it > end - 4)
-    return nan;
-  if (!scanInt(it, it + 2, tzh))
-    return nan;
-  tzh *= sign;
-  if (!scanInt(it, it + 2, tzm))
+  // Timezone MUST be set.
+  if (it == end) {
     return nan;
-  tzm *= sign;
+  }
+  // Hour and minute of timezone adjustment.
+  if (consume(u':') && it == end) {
+    // See if timezone is '-:' or '+:'
```

Comment:
Why is `-:` a valid timezone adjustment? Is the implication that the adjustment is `-00:00`, but because the numbers were omitted, we need to add the `localTZA()` anyway?

Reply:
I'll revert this changes.